control-userlayer/src/main/java/org/oskari/control/userlayer/EditUserLayerHandler.java

```diff
@@ -0,0 +1,91 @@
+package fi.nls.oskari.control.data;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import fi.mml.map.mapwindow.util.OskariLayerWorker;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionDeniedException;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionHandler;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.ActionParamsException;
+import fi.nls.oskari.domain.map.userlayer.UserLayer;
+import fi.nls.oskari.domain.map.userlayer.UserLayerStyle;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.util.ConversionHelper;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.ResponseHelper;
+import org.oskari.map.userlayer.service.UserLayerDataService;
+import org.oskari.map.userlayer.service.UserLayerDbService;
+
+/**
+ * Expects to get layer id as http parameter "id".
+ */
+@OskariActionRoute("EditUserLayer")
+public class EditUserLayerHandler extends ActionHandler {
+    private final static String PARAM_ID = "id";
+    private static final String PARAM_DESC = "desc";
+    private static final String PARAM_NAME = "name";
+    private static final String PARAM_SOURCE = "source";
+    private static final String PARAM_STYLE = "style";
+    
+    private UserLayerDbService userLayerDbService = null;
+    private final UserLayerDataService userlayerService = new UserLayerDataService();
+    
+    public void setUserLayerDbService(final UserLayerDbService service) {
+        userLayerDbService = service;
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        if(userLayerDbService == null) {
+            setUserLayerDbService(OskariComponentManager.getComponentOfType(UserLayerDbService.class));
+        }
+    }
+
+    @Override
+    public void handleAction(ActionParameters params) throws ActionException {
+        params.requireLoggedInUser();
+
+        final long id = ConversionHelper.getLong(params.getHttpParam(PARAM_ID), -1);
+
+        if(id == -1) {
+            throw new ActionParamsException("Parameter missing or non-numeric: " + PARAM_ID + "=" + params.getHttpParam(PARAM_ID));
+        }
+        
+        final UserLayer userLayer = userLayerDbService.getUserLayerById(id);
+        final UserLayerStyle style = new UserLayerStyle();
+
+        if(userLayer == null) {
+            throw new ActionParamsException("Userlayer id doesn't exist: " + id);
+        }
+        if(!userLayer.isOwnedBy(params.getUser().getUuid())) {
+            throw new ActionDeniedException("Userlayer belongs to another user");
+        }
+               
+        userLayer.setLayer_name(params.getHttpParam(PARAM_NAME));
```

Comment:
If these parameters are not set do we want to replace them? If not could call `userLayer.setLayer_name(params.getHttpParam(PARAM_NAME, userLayer.getLayer_name());`

Reply:
Maybe this is better, since then it would be impossible for the frontend to set the value to null

Reply:
Things will probably break with null so prefer getRequiredParam() or default to previous name.